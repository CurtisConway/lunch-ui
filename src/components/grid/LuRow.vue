<template>
  <component
    :is="tag"
    :class="defaultClassList"
  >
    <slot></slot>
  </component>
</template>

<script>
export default {
  name: 'LuRow',
  props: {
    tag: {
      type: String,
      default: () => 'div',
    },
    reverse: {
      type: Boolean,
      default: () => false,
    },
    wrap: {
      type: Boolean,
      default: () => false,
    },
    vAlign: {
      type: String,
      default: () => 'center',
    },
    hAlign: {
      type: String,
      default: () => 'left',
    },
  },
  computed: {
    defaultClassList() {
      return [
        {
          flex: true,
          'flex-row': true,
          'flex-reverse': this.reverse,
          'flex-wrap': this.wrap,
        },
        this.horizontalAlignmentClasses,
        this.verticalAlignmentClasses,
      ];
    },
    horizontalAlignmentClasses() {
      return {
        'align-h-left': this.hAlign === 'left',
        'align-h-center': this.hAlign === 'center',
        'align-h-right': this.hAlign === 'right',
      };
    },
    verticalAlignmentClasses() {
      return {
        'align-v-top': this.vAlign === 'top',
        'align-v-center': this.vAlign === 'center',
        'align-v-bottom': this.vAlign === 'bottom',
      };
    },
  },
};
</script>
