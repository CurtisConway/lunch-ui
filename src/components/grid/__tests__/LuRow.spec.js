import Vue from 'vue';
import Router from 'vue-router';

import LuRow from '../LuRow.vue';

import {
  createLocalVue,
  mount,
} from '@vue/test-utils';
import { compileToFunctions } from 'vue-template-compiler'

describe('LuRow.vue', () => {
  let mountFunction;
  let router = Router;
  let localVue = typeof Vue;

  beforeEach(() => {
    router = new Router();
    localVue = createLocalVue();
    localVue.use(Router);

    mountFunction = (options = {}) => {
      return mount(LuRow, {
        localVue,
        router,
        ...options,
      });
    }
  })

  it('should render component and match snapshot', () => {
    expect(mountFunction().html()).toMatchSnapshot();
  });

  it('should render component with a tag prop and match snapshot', () => {
    expect(mountFunction({
      propsData: {
        tag: 'main',
      },
    }).html()).toMatchSnapshot();
  });

  it('should render component with a reverse prop and match snapshot', () => {
    expect(mountFunction({
      propsData: {
        reverse: true,
      },
    }).html()).toMatchSnapshot();
  });

  it('should render component with a wrap prop and match snapshot', () => {
    expect(mountFunction({
      propsData: {
        wrap: true,
      },
    }).html()).toMatchSnapshot();
  });

  it('should render component with a hAlign prop and match snapshot', () => {
    expect(mountFunction({
      propsData: {
        hAlign: 'left',
      },
    }).html()).toMatchSnapshot();
  });

  it('should render component with a vAlign prop and match snapshot', () => {
    expect(mountFunction({
      propsData: {
        vAlign: 'top',
      },
    }).html()).toMatchSnapshot();
  });

  it('should have the flex and flex-row classes', async () => {
    const wrapper = mountFunction({
      propsData: {},
    });

    expect(wrapper.classes('flex')).toBe(true);
    expect(wrapper.classes('flex-row')).toBe(true);
  });

  it('should render the tag passed in', async () => {
    const tag = 'main';
    const wrapper = mountFunction({
      propsData: {
        tag
      },
    });

    expect(wrapper.is(tag)).toBe(true);
  });

  it('should have the flex-reverse class with the reverse prop', async () => {
    const wrapper = mountFunction({
      propsData: {
        reverse: true,
      },
    });

    expect(wrapper.classes('flex-reverse')).toBe(true);
  });

  it('should have the flex-wrap class with the wrap prop', async () => {
    const wrapper = mountFunction({
      propsData: {
        wrap: true,
      },
    });

    expect(wrapper.classes('flex-wrap')).toBe(true);
  });

  it('should have the right align-h-* class with the alignH prop', async () => {
    const left = mountFunction({
      propsData: {
        hAlign: 'left',
      },
    });
    const center = mountFunction({
      propsData: {
        hAlign: 'center',
      },
    });
    const right = mountFunction({
      propsData: {
        hAlign: 'right',
      },
    });

    expect(left.classes('align-h-left')).toBe(true);
    expect(center.classes('align-h-center')).toBe(true);
    expect(right.classes('align-h-right')).toBe(true);
  });

  it('should have the right align-v-* class with the alignH prop', async () => {
    const top = mountFunction({
      propsData: {
        vAlign: 'top',
      },
    });
    const center = mountFunction({
      propsData: {
        vAlign: 'center',
      },
    });
    const bottom = mountFunction({
      propsData: {
        vAlign: 'bottom',
      },
    });

    expect(top.classes('align-v-top')).toBe(true);
    expect(center.classes('align-v-center')).toBe(true);
    expect(bottom.classes('align-v-bottom')).toBe(true);
  });
});
