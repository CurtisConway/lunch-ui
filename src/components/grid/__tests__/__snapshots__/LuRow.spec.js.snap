// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuRow.vue should render component and match snapshot 1`] = `<div class="flex flex-row flex-wrap"></div>`;

exports[`LuRow.vue should render component with a hAlign prop and match snapshot 1`] = `<div class="flex flex-row flex-wrap align-h-left"></div>`;

exports[`LuRow.vue should render component with a reverse prop and match snapshot 1`] = `<div class="flex flex-row flex-reverse flex-wrap"></div>`;

exports[`LuRow.vue should render component with a tag prop and match snapshot 1`] = `<main class="flex flex-row flex-wrap"></main>`;

exports[`LuRow.vue should render component with a vAlign prop and match snapshot 1`] = `<div class="flex flex-row flex-wrap align-v-top"></div>`;

exports[`LuRow.vue should render component with a wrap prop and match snapshot 1`] = `<div class="flex flex-row flex-wrap"></div>`;
